pymc/distributions/distribution.py

```diff
@@ -150,35 +149,34 @@ def fn(*args, **kwargs):
     return fn
 
 
-def _make_rv_and_resize_shape(
+def _make_rv_and_resize_shape_from_dims(
     *,
     cls,
-    dims: Optional[Dims],
+    dims: Optional[StrongDims],
     model,
     observed,
     args,
     **kwargs,
-) -> Tuple[Variable, Optional[WeakDims], Optional[Union[np.ndarray, Variable]], StrongShape]:
-    """Creates the RV and processes dims or observed to determine a resize shape."""
-    # Create the RV without dims information, because that's not something tracked at the Aesara level.
-    # If necessary we'll later replicate to a different size implied by already known dims.
+) -> Tuple[Variable, StrongShape]:
+    """Creates the RV, possibly using dims or observed to determine a resize shape (if needed)."""
+    resize_shape_from_dims = None
+    size_or_shape = kwargs.get("size") or kwargs.get("shape")
+
+    # Preference is given to size or shape. If not specified, we rely on dims and
+    # finally, observed, to determine the shape of the variable. Because dims can be
+    # specified on the fly, we need a two-step process where we first create the RV
+    # without dims information and then resize it.
+    if not size_or_shape and observed is not None:
+        kwargs = kwargs.copy()
+        kwargs["shape"] = tuple(observed.shape)
+
+    # Create the RV without dims information
     rv_out = cls.dist(*args, **kwargs)
-    ndim_actual = rv_out.ndim
-    resize_shape = None
-
-    # # `dims` are only available with this API, because `.dist()` can be used
-    # # without a modelcontext and dims are not tracked at the Aesara level.
-    dims = convert_dims(dims)
-    dims_can_resize = kwargs.get("shape", None) is None and kwargs.get("size", None) is None
-    if dims is not None:
-        if dims_can_resize:
-            resize_shape, dims = resize_from_dims(dims, ndim_actual, model)
-        elif Ellipsis in dims:
-            # Replace ... with None entries to match the actual dimensionality.
-            dims = (*dims[:-1], *[None] * ndim_actual)[:ndim_actual]
-    elif observed is not None:
-        resize_shape, observed = resize_from_observed(observed, ndim_actual)
-    return rv_out, dims, observed, resize_shape
+
+    if not size_or_shape and dims is not None:
```

Comment:
Couldn't we use this spot to do some error checking? If a size or shape is specified and dims is also specified, then we could check if they are consistent, and throw an error if they are not. So we wouldn't have to say something "shape takes precedent", but just: "If you specify both they must say the same thing". Otherwise I think we would get an error when we try to export a trace to arviz, right? And we'd rather get the error at definition, than getting it after sampling? Or is there a check for this somewhere else already?

We could then add a test like this:

```python
with pytest.raises(ValueError):
    with pm.Model(coords={"a": [1, 2]}) as model:
        pm.Normal("y", shape=3, dims="a")
```

Reply:
The question about precedence in this comment is not related to error checking but what exactly goes into the RV size argument. The reason we give preference to `size`/`shape` is because it can be an arbitrary expression and so is more powerful in principle than dims.

This PR keeps the same level of no-error checking as before. The direct testa in `test_shape_handling` mention this in their comments. 

I would live such discussion outside of this PR unless it feels very relevant.

Reply:
Sure, it isn't really related to the issue in the PR.

I'll open a separate PR about that, I think we can at least do some checks if the shape/size is an integer. Has been too long anyway since I did my last pymc PR. :-)